sebs/fission/fission.py

```diff
@@ -0,0 +1,373 @@
+            stop_minikube()
+        self.storage.storage_container.kill()
+        logging.info("Minio stopped")
+
+    def get_storage(self, replace_existing: bool = False) -> PersistentStorage:
+        self.storage = Minio(self.docker_client)
+        return self.storage
+
+    def initialize(self, config: Dict[str, str] = None):
+        if config is None:
+            config = {}
+
+        check_if_minikube_installed()
+        check_if_k8s_installed()
+        check_if_helm_installed()
+        run_minikube()
+        Fission.add_port_forwarding()
+        sleep(5)
+
+    def package_code(self, benchmark: Benchmark) -> Tuple[str, int]:
+
+        benchmark.build()
+
+        CONFIG_FILES = {
+            "python": [
+                "handler.py",
+                "requirements.txt",
+                ".python_packages",
+                "build.sh",
+            ],
+            "nodejs": ["handler.js", "package.json", "node_modules"],
+        }
+        directory = benchmark.code_location
+        package_config = CONFIG_FILES[benchmark.language_name]
+        function_dir = os.path.join(directory, "function")
+        os.makedirs(function_dir)
+        minioConfig = open("./code/minioConfig.json", "w+")
+        minioConfigJson = {
+            "access_key": self.storage.access_key,
+            "secret_key": self.storage.secret_key,
+            "url": self.storage.url,
+        }
+        minioConfig.write(json.dumps(minioConfigJson))
+        minioConfig.close()
+        scriptPath = os.path.join(directory, "build.sh")
+        self.shouldCallBuilder = True
+        f = open(scriptPath, "w+")
+        f.write(
+            "#!/bin/sh\npip3 install -r ${SRC_PKG}/requirements.txt -t \
+${SRC_PKG} && cp -r ${SRC_PKG} ${DEPLOY_PKG}"
+        )
+        f.close()
+        subprocess.run(["chmod", "+x", scriptPath])
+        for file in os.listdir(directory):
+            if file not in package_config:
+                file = os.path.join(directory, file)
+                shutil.move(file, function_dir)
+        os.chdir(directory)
+        subprocess.run(
+            "zip -r {}.zip ./".format(benchmark.benchmark).split(),
+            stdout=subprocess.DEVNULL,
+        )
+        benchmark_archive = "{}.zip".format(os.path.join(directory, benchmark.benchmark))
+        logging.info("Created {} archive".format(benchmark_archive))
+        bytes_size = os.path.getsize(benchmark_archive)
+        return benchmark_archive, bytes_size
+
+    def update_function(self, name: str, env_name: str, code_path: str):
+        self.create_function(name, env_name, code_path)
+
+    def create_env_if_needed(self, name: str, image: str, builder: str):
+        self.envName = name
+        try:
+            fission_env_list = subprocess.run(
+                "fission env list ".split(),
+                stdout=subprocess.PIPE,
+                stderr=subprocess.DEVNULL,
+            )
+            subprocess.run(
+                f"grep {name}".split(),
```

Comment:
This would be easier to do with just a Python loop parsing the output, instead of calling Linux process again.